Makefile

```
@@ -308,11 +308,21 @@ define find_all_clojure_files
 $$(comm -23 <(sort <(git ls-files --cached --others --exclude-standard)) <(sort <(git ls-files --deleted)) | grep -e \.clj$$ -e \.cljs$$ -e \.cljc$$ -e \.edn)
 endef
 
+# Lint Clojure files with clj-kondo.
+#
+# Args:
+#   $1: When equal to true, print warnings.
+define lint_clojure_files
+	$(if $(filter true,$1), \
+			clj-kondo --config .clj-kondo/config.edn --cache false --fail-level error --lint src, \
+			clj-kondo --config .clj-kondo/config.edn --cache false --fail-level error --lint src | grep -v ': warning: ')
+endef
+
 lint: export TARGET := clojure
 lint: ##@test Run code style checks
 	@sh scripts/lint-re-frame-in-quo-components.sh && \
-	sh scripts/lint-old-quo-usage.sh \
-	clj-kondo --config .clj-kondo/config.edn --cache false --fail-level error --lint src && \
+	sh scripts/lint-old-quo-usage.sh && \
+	$(call lint_clojure_files, $(CLJ_LINTER_PRINT_WARNINGS)) && \
```

Can't we just inline it?
```suggestion
	clj-kondo --config .clj-kondo/config.edn --cache false --fail-level error --lint src | grep -v ': warning: ') $(ifneq ($(CLJ_LINTER_PRINT_WARNINGS), true)| grep -v ': warning: ' endif)
```

---

Most definitely.

---

Thanks for the suggestion @jakubgs.

I couldn't inline the call with the suggested code. I tried a few different incantations with `ifneq`, but nothing worked.

Using `filter` in combination with `if` as a one-liner gave the correct results, so I think it's fine. Commit https://github.com/status-im/status-mobile/pull/17491/commits/2f2f4dc624995df3ad191bd8eb3ade054c569a4c

---

I see, because `if` is a function and `ifeq`/`ifneq` are directives: https://stackoverflow.com/a/9008922